providers/x509.rb

```diff
@@ -67,6 +69,16 @@ def cert
   @cert ||= OpenSSL::X509::Certificate.new
 end
 
+def cert_file
+  @cert_file ||= if Pathname.new(new_resource.name).absolute?
+                   new_resource.name
+                 else
+                   cert_dir = new_resource.cert_dir || node['openssl']['default_x509_dir']
```

Comment:
In general we don't want to have a resource that's dependent on a cookbook attribute. Our long term goal in this resource is to move it into chef and once we do that we won't have this attribute anymore.